token-lending/program/src/state/reserve.rs

```diff
@@ -290,10 +291,24 @@ impl Reserve {
             }
         }
 
+        // calc the bonus amount
+        let base_amount = Decimal::from(withdraw_amount).try_div(bonus_rate)?;
+        let bonus_amount = Decimal::from(withdraw_amount).try_sub(base_amount)?;
+
+        // protocol takes a percentage of the bonus
+        let mut protocol_fee_amount = bonus_amount.try_mul(protocol_fee_rate)?.try_floor_u64()?;
+        // minimum protocol fee
+        if protocol_fee_amount == 0 {
+            protocol_fee_amount = 1;
+        }
+
+        let withdraw_amount = withdraw_amount.saturating_sub(protocol_fee_amount);
```

Comment:
why saturating here? can you give an example where why we wouldn't just want try_sub